### PR TITLE
add anonymization 

### PR DESCRIPTION
**Pull request by romaissaMe**

### CODE REVIEW

File: anonymize.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from PyPDF2 import PdfReader
+
+reader=PdfReader("meta_projet_final.pdf")
```
Comment (achab) [inline anonymize.py:3]: can you add meta_projet_final.pdf to the repo?

File: anonymize.py [hunk 2]
```diff
@@ -1,11 +1,49 @@
-from PyPDF2 import PdfReader
+import json
```
Comment (achab) [inline anonymize.py:1]: you don't use this library, please remove it

File: anonymize.py [hunk 3]
```diff
@@ -1,11 +1,49 @@
...
+from presidio_anonymizer import AnonymizerEngine
+from presidio_analyzer import AnalyzerEngine
+from presidio_anonymizer.entities import RecognizerResult, OperatorConfig
```
Comment (achab) [inline anonymize.py:4]: you don't use `RecognizerResult`, please remove it

File: anonymize.py [hunk 4]
```diff
@@ -1,11 +1,49 @@
...
-    page=reader.pages[i]
-    text += page.extract_text()
+with open("info.txt",'r',encoding='utf-8') as file:
+    text=file.read()
 
-with open("text_file",'w',encoding='utf-8') as file:
-    file.write(text)
\ No newline at end of file
+# 1/ Analyzer 
+#text = "I am louis, my second name is James Bond, my phone number is 07816093423"
```
Comment (achab) [inline anonymize.py:13]: remove those lines

File: anonymize.py [hunk 5]
```diff
@@ -1,11 +1,49 @@
...
+analyzer = AnalyzerEngine()
+analyzer_results = analyzer.analyze(text=text, language="en")
+
+# Analyzer output
+print("analyzer results")
+print(analyzer_results)
```
Comment (achab) [inline anonymize.py:19]: printing is good for testing, but it's cleaner to remove it from the final file

File: anonymize.py [hunk 6]
```diff
@@ -1,11 +1,49 @@
...
+from presidio_anonymizer.entities import RecognizerResult, OperatorConfig
 
-reader=PdfReader("meta_projet_final.pdf")
 
+#read txt file
 text=''
-for i in range(len(reader.pages)):
-    page=reader.pages[i]
-    text += page.extract_text()
+with open("info.txt",'r',encoding='utf-8') as file:
```
Comment (achab) [inline anonymize.py:9]: does it work if you remove `encoding="utf-8"` here? encoding is platform-dependent (which means its different for ubuntu and windows), so I cannot test easily on my machine (ubuntu) if it works on windows

File: info.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+I am louis, my second name is James Bond, my phone number is 07816093423
```
Comment (achab) [inline info.txt:1]: Two comments here:
- can you rename the file to `text.txt` (you will need to rename the file locally, and run `git add text.txt`)
- can you add more personal information to the text? like an physical address, an email address and a url?

**Comment by achab** [comment]
Very good first PR @romaissaMe! I've added a few comments to clean the code, but most of it is very good. Could you also add some text to the file `README.md` to explain what the file `anonymize.py` does?

File: anonymize.py (hunk 2 above)
Comment (romaissaMe) [reply anonymize.py:1]: weird this doesn't appear in my code or the code in the repo!

File: anonymize.py (hunk 6 above)
Comment (romaissaMe) [reply anonymize.py:9]: yes it works , I thought it won't accept french characters without utf-8 but it turns that it does!

File: anonymize.py (hunk 2 above)
Comment (achab) [reply anonymize.py]: you mean locally? if you want to show the difference between the last commit and your current code, you can run `git diff`

File: requirements.txt [hunk 8]
```diff
@@ -0,0 +1,2 @@
+presidio_analyzer
+presidio_anonymizer
```
Comment (achab) [inline requirements.txt:2]: please add version to these packages? you can look at the version on your local machine like that:

`pip freeze | grep presidio_analyzer`

**Review by achab — APPROVED** [review]
I've put two last comments. Once they are addressed, feel free to merge the PR by clicking on "Squash and merge" button. Congrats for this first PR!

**Pull request merged**